app/routes/errors.py

```diff
@@ -41,11 +44,14 @@ def _render_error_page(status_code, template=None):
     )
 
 
-@errors_blueprint.app_errorhandler(401)
-@errors_blueprint.app_errorhandler(CSRFError)
 @errors_blueprint.app_errorhandler(NoTokenException)
-def unauthorized(error=None):
+@errors_blueprint.app_errorhandler(CSRFError)
+@errors_blueprint.app_errorhandler(NoQuestionnaireStateException)
+@errors_blueprint.app_errorhandler(401)
+def no_cookie(error=None):
     log_error(error, 401)
+    if not cookie_session:
+        return _render_error_page(401, "no-cookie")
     return _render_error_page(401, "session-expired")
```

Comment:
Not sure if this is working as expected. If you now go to a questionnaire endpoint, the error you get first time is not the same one if you refresh the page because you don't have a cookie on the first request, but then we sent one back, so on a refresh you do have a cookie.

Reply:
I think it should exist for any endpoints that allow a POST. I've had a look at the Flask docs and it looks like we can use [request.url_rule.methods](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.url_rule) for this. 

As all templates go via the `render_template` method we could add an `include_csrf_token` template param using something like:

```python
include_csrf_token=True if "POST" in request.url_rule.methods else False
```

Then make the inclusion of the csrf token in `_base.html` conditional on this.